refactor: preheating only in test handler

benesjan opened the pull request:
Addresses https://linear.app/aztec-labs/issue/TMNT-60/ignition-rollup-contracts-internal-review#comment-c454ca97


AI one-shotted this:
<img width="987" height="994" alt="image" src="https://github.com/user-attachments/assets/61615383-a4d3-4e0d-b6a8-a8cdd0909260" />

Pretty cool

benesjan commented:
* **#16404** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16404?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16404?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

LHerskind commented:
Will push a tiny thing on top. We don't need to have it as externally callable, it can just be internal and called as part of the constructor, then you can also avoid some of the checks for preheated and get rid of a few functions. Also it don't update the toml so it breaks the ges benchmark as the data is not getting collected

LHerskind committed:
refactor: preheating only in test handler

Addresses https://linear.app/aztec-labs/issue/TMNT-60/ignition-rollup-contracts-internal-review#comment-c454ca97

AI one-shotted this:
<img width="987" height="994" alt="image" src="https://github.com/user-attachments/assets/61615383-a4d3-4e0d-b6a8-a8cdd0909260" />

Pretty cool
